backend/logic/courses/courses.js

```diff
@@ -261,12 +261,104 @@ var updateCourse = async (course, newTitle, newTags) => {
   }
 };
 
+/**
+ * Assign a course to a curriculum
+ * @author Yufei Liu
+ * @param {string} curriculumType,{string} courseCode,{string} curriculum
+ * @returns true if successful
+ * @throws error if curriculumType, courseCode or curriculum dose not exist
+ *         course already assigned a curriculum
+ *
+ */
+var assignCourseToCurriculum = async (
+  curriculumType,
+  courseCode,
+  curriculum
+) => {
+  let conn = await mysql.getNewConnection();
+
```

Comment:
Maybe check if courseCode is in a valid format before proceeding with codes below